### ISSUE TITLE
Metrics are incorrect despite correct prediction?

### ISSUE THREAD

**Issue by Stupack** [ThilinaRajapakse/simpletransformers#714]
Hi, I was experimenting with the following code:

```
from simpletransformers.ner.ner_model import NERModel

model_args = {
    "num_train_epochs": 20,
    "reprocess_inputs" :False
}

train_data = [
    [0, 'what', 'O'], [0, 'is', 'O'], [0, 'the', 'O'], [0, 'phone', 'OBJ'], [0, 'number', 'O'], [0, 'and', 'O'], [0,'postcode', 'OBJ'], [0, 'of', 'O'], [0,'a', 'O'], [0, 'cheap', 'OBJ'], [0, 'restaurant', 'OBJ'], [0, 'in', 'O'], [0, 'the', 'O'], [0, 'east', 'OBJ'], [0, 'part', 'OBJ'], [0, 'of', 'O'], [0, 'town', 'OBJ'],[0, 'town', 'OBJ'], [0, '?', 'O']
]
train_df = pd.DataFrame(train_data, columns=['sentence_id', 'words', 'labels'])
# Create a NERModel
model = NERModel('bert', 'bert-base-cased', use_cuda=False, labels=['OBJ', 'O', 'ACT'], args=model_args)


# Train the model
model.train_model(train_df)

# Evaluate the model
result, model_outputs, predictions = model.eval_model(train_df)
print(result)

```
`{'eval_loss': 0.0023626224137842655, 'precision': 0, 'recall': 0, 'f1_score': 0}`

I made it to overfit the training data and evaluate in the same data but the results are always zero. I dont understand what's wrong.

I checked the output:

`predictions, raw_outputs = model.predict(["what is the phone number and postcode of a cheap restaurant in the east part of town"])`
```
[[{'what': 'O'},
  {'is': 'O'},
  {'the': 'O'},
  {'phone': 'OBJ'},
  {'number': 'O'},
  {'and': 'O'},
  {'postcode': 'OBJ'},
  {'of': 'O'},
  {'a': 'O'},
  {'cheap': 'OBJ'},
  {'restaurant': 'OBJ'},
  {'in': 'O'},
  {'the': 'O'},
  {'east': 'OBJ'},
  {'part': 'OBJ'},
  {'of': 'O'},
  {'town': 'OBJ'}]]
```

**Comment by ThilinaRajapakse** [comment]
There's definitely something odd going on with the seqeval metric function.

```python
>>> y_true = [['O', 'O', 'O', 'OBJ', 'O', 'O', 'OBJ', 'O', 'O', 'OBJ', 'OBJ', 'O', 'O', 'OBJ', 'OBJ', 'O', 'OBJ', 'OBJ', 'O']]
>>> y_pred = [['O', 'O', 'O', 'OBJ', 'O', 'O', 'OBJ', 'O', 'O', 'OBJ', 'OBJ', 'O', 'O', 'OBJ', 'OBJ', 'O', 'OBJ', 'OBJ', 'O']]
>>> precision_score(y_true, y_pred)
0
>>> y_pred = [['O', 'O', 'O', 'OBJ', 'O', 'O', 'OBJ', 'O', 'O', 'OBJ', 'OBJ', 'O', 'O', 'OBJ', 'OBJ', 'O', 'OBJ', 'OBJ']]
>>> y_true = y_pred
>>> y_pred = [['O', 'O', 'O', 'OBJ', 'O', 'O', 'OBJ', 'O', 'O', 'OBJ', 'OBJ', 'O', 'O', 'OBJ', 'OBJ', 'O', 'OBJ', 'OBJ']]
>>> precision_score(y_true, y_pred)
0
>>> y_pred = [['OBJ', 'OBJ', 'OBJ', 'OBJ', 'OBJ', 'OBJ', 'OBJ', 'OBJ']]
>>> y_true = y_pred
>>> precision_score(y_true, y_pred)
0
>>> y_pred = [['OBJ']]
>>> y_true = y_pred
>>> precision_score(y_true, y_pred)
0
>>> y_pred = [['J']]
>>> y_true = y_pred
>>> precision_score(y_true, y_pred)
1.0
>>> y_pred = [['OJ']]
>>> y_true = y_pred
>>> precision_score(y_true, y_pred)
0
>>> y_pred = [['O']]
>>> y_true = y_pred
>>> precision_score(y_true, y_pred)
0
>>> 

```

I am stumped as well.

**Comment by moritzblum** [comment]
The scores are probably wrong because the labelling is not following a supported schema. According to [seqeval](https://github.com/chakki-works/seqeval), the following formats are supported: IOB1, IOB2, IOE1, IOE2, IOBES. Inserting B and I should lead to correct scores:
```
>>> y_true = [['O', 'O', 'O', 'B-OBJ', 'O', 'O', 'I-OBJ', 'O', 'O', 'B-OBJ', 'I-OBJ', 'O', 'O', 'B-OBJ', 'I-OBJ', 'O', 'I-OBJ', 'I-OBJ', 'O']]
>>> y_pred = y_true
>>> precision_score(y_true, y_true)
1.0
```
